### PR TITLE
Feature: Internal debugger step into and memory search. 

### PR DESCRIPTION
**Pull request by maximilien-noal**
### Description of Changes

Introduces fixes, and some new features to the internal debugger. The first step was to make it have the same feature set as the GDB debugger (memory search was missing) The next step was to add a 'step into' feature. Other improvements will come after this one. 

### Rationale behind Changes

The internal debugger should help with reverse engineering Dune. And of course, if it applies to Dune, it works with other games that the emulator can run.

### Suggested Testing Steps

Use it with your favorite reverse engineering project.

### CODE REVIEW

**Review by kevinferrare — APPROVED** [review]

File: src/Spice86.Core/Emulator/CPU/Stack.cs [hunk 1]
```diff
@@ -164,7 +168,7 @@ public string PeekWindow(int range = 8) {
             if (i == PhysicalAddress) {
                 sb.Append('*');
             }
-            sb.Append("[0x").AppendFormat("{0:X6}", i).Append("] 0x").AppendFormat("{0:X4}", _memory.UInt16[i]).AppendLine();
+            sb.Append("[0x").Append($"{i:X6}").Append("] 0x").Append($"{_memory.UInt16[i]:X4}").AppendLine();
```
Comment (JorisVanEijden) [inline src/Spice86.Core/Emulator/CPU/Stack.cs:171]:
```suggestion
            sb.AppendLine($"[0x{i:X6}] 0x{_memory.UInt16[i]:X4}");
```

File: src/Spice86.Core/Emulator/Devices/Timer/CounterConfiguratorFactory.cs [hunk 2]
```diff
@@ -5,18 +5,32 @@ namespace Spice86.Core.Emulator.Devices.Timer;
...
         _loggerService = loggerService;
+        _pauseHandler = pauseHandler;
         _configuration = configuration;
     }
 
-    public CounterActivator InstanciateCounterActivator(State state) {
+    /// <summary>
+    /// Creates a new instance of one of the implementations of <see cref="CounterActivator"/> class based on the emulator configuration.
+    /// </summary>
+    public CounterActivator InstanciateCounterActivator() {
```
Comment (JorisVanEijden) [inline src/Spice86.Core/Emulator/Devices/Timer/CounterConfiguratorFactory.cs:33]:
```suggestion
    public CounterActivator InstantiateCounterActivator() {
```

File: src/Spice86.Shared/Utils/ConvertUtils.cs [hunk 3]
```diff
@@ -99,12 +102,36 @@ public static uint BytesToInt32(byte[] data, int start) {
...
         return res;
     }
+    
+    /// <summary>
+    /// Tries to convert a hexadecimal string to a byte array.
+    /// </summary>
+    /// <param name="valueString">The hexadecimal string to convert.</param>
+    /// <param name="bytes">The byte array representation of the hexadecimal string.</param>
+    /// <returns><c>False</c> if the conversion fails, <c>True</c> otherwise</returns>
+    public static bool TryParseHexToByteArray(string valueString, [NotNullWhen(true)] out byte[]? bytes) {
```
Comment (JorisVanEijden) [inline src/Spice86.Shared/Utils/ConvertUtils.cs:118]: How is this different from [Convert.FromHexString](https://learn.microsoft.com/en-us/dotnet/api/system.convert.fromhexstring?view=net-8.0)?

Comment (maximilien-noal) [reply src/Spice86.Shared/Utils/ConvertUtils.cs:118]: It's used to prevent an exception if the UI input from the user is not recognized.

**Pull request merged**